src/registries/npm-registry.js

```diff
@@ -72,10 +73,22 @@ export default class NpmRegistry extends Registry {
     return name.replace('/', '%2f');
   }
 
+  getRequestUrl(registry: string, pathname: string): string {
+    const isPathnameUrl = pathname.match(/^https?:/);
+
+    if (isPathnameUrl) {
+      return pathname;
+    } else {
+      return url.resolve(registry, pathname);
+    }
+  }
+
   request(pathname: string, opts?: RegistryRequestOptions = {}, packageName: ?string): Promise<*> {
     const registry = this.getRegistry(packageName || pathname);
-    const requestUrl = url.resolve(registry, pathname);
+    const requestUrl = this.getRequestUrl(registry, pathname);
+
     const alwaysAuth = this.getRegistryOrGlobalOption(registry, 'always-auth');
+    const customHostSuffix = this.getRegistryOrGlobalOption(registry, 'custom-host-suffix');
```

Comment:
I will have to move the `isRequestToRegistry` inside the `NPMRegistry` class since it will needs access to the config to read the 'custom-host-suffix`.